Stable 1.5

davidbartonau opened the pull request:
@Vlad1mir-D @dragonroot I hope you are both well at the moment with everything going on.

I've been very slow doing what I promised, but I have hammered the latest version in the hopes that a 1.5 can come out.  It's been tested in Ubuntu 18.04 and 20.04 with 10,000s of backups and 100,000s of restores.

I picked up one very minor bug, if you create a backup with a folder called `backups`, it tries to find an info file there and fails.  I am guessing that issue has been there "forever", and it seems no-one is likely to care.

I added a note about borg as well, because I think it is quite popular but has some pretty bad performance characteristics compared to zbackup for the use cases I tested.

davidbartonau committed:
Regression testing of zbackup.

davidbartonau committed:
Add regression stress test

davidbartonau committed:
Added notes about the testing in the readme.

sanmai commented:
I wonder if you want to mention `restic` too.

davidbartonau commented:
@sanmai I recall looking into restic some years back, but I cannot remember for the life of me why I ended up not using it.  Do you know anything about it's good / bad points?

sanmai commented:
I can tell that it's fast, and fairly usable as a replacement for a tool like Borg or Attic.

I can't compare any of these with ZBackup TBH, because ZBackup does one or two things best, where these other tools do this, that, and the other thing. E.g. they dab as a tar-like program, where for ZBackup you have to use actual tar (a good thing!).  ZBackup is a lot more Unix-y compared to them.

adriaandegroot committed:
Remove use_count() from the smart pointer

- method is unused in the zbackup codebase
- returns the **pointer** when it should be returning the value
  it points to, e.g. , so that was wrong always.
  Causes build failures with modern C++ compilers (clang9
  for instance)

adriaandegroot committed:
Require C++11

- Update CMake version to one that is still supported
- Bump C++ version to 11, since all current compilers support that
- Give sptr.hh the nullptr treatment (instead of 0 -- this is
  one of the nice things C++11 gives us)

adriaandegroot committed:
Need unistd for unlink()

adriaandegroot committed:
Globbing for sources is considered poor form

adriaandegroot committed:
Turn BUSE into an option (because it's Linux-only)

- option WITH_BUSE at CMake level
- a handful of #ifdefs to take out the relevant consumer code

davidbartonau committed:
Manually merge pull request #153 in zbackup from adriaandegroot/zbackup

davidbartonau committed:
Manually merge pull request #154 in zbackup from adriaandegroot/zbackup

davidbartonau committed:
Manually merge pull request #155 in zbackup from adriaandegroot/zbackup

davidbartonau committed:
Manually merge pull request #156 in zbackup from adriaandegroot/zbackup

davidbartonau committed:
Make automated testing simpler

Commit:
Replace obsolete version of theSetTotalBytesLimit that used the warning setting

mwt commented:
As far as I can tell, Debian has packaged this 1.5 branch since Bookworm. So, it is pretty well tested at this point. It would probably make sense to merge this if @Vlad1mir-D @dragonroot is available.
